### PR TITLE
Added log4j2 instead logback

### PR DESCRIPTION
**Pull request by artem-v**
Covered issue: https://github.com/scalecube/scalecube/issues/61

### CODE REVIEW

File: transport/pom.xml [hunk 1]
```diff
@@ -59,6 +59,16 @@
             <artifactId>slf4j-api</artifactId>
         </dependency>
         <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
```
Comment (antonkharenko) [inline transport/pom.xml]: Lets add logger implementation dependencies only to testlib

File: testlib/src/main/resources/log4j2-test.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <Appenders>
+        <Console name="console" target="SYSTEM_OUT">
+            <PatternLayout>
+                <pattern>%level{length=1} %date{MMdd-HHmm:ss,SSS} %logger{1.} %message [%thread]%n</pattern>
+            </PatternLayout>
+        </Console>
+    </Appenders>
+
+    <Loggers>
+        <Logger name="io.netty" level="info"/>
```
Comment (antonkharenko) [inline testlib/src/main/resources/log4j2-test.xml]: Missed - <logger name="io.scalecube.transport" level="info"/> it was not by mistake

File: transport/pom.xml (hunk 1 above)
Comment (artem-v) [inline transport/pom.xml]: @antonkharenko in the last commit I have added log4j-test cfg in every module with test code. Dependency on log4j marked as ``optional``.

**Comment by antonkharenko** [comment]
@artem-v This solution looks unnecessary complex. Currently there is only logback.xml and dependency in testlib module and this it. In proposed PR config and dependencies are copy-pasted at several modules (the ones which do not require at all dependency on logger). I tend to reject this PR since there is no difference what logger is used by tests at the moment so I prefer to stay with the simplest possible solution.

**Comment by artem-v** [comment]
@antonkharenko now I put dep on log4j only in testlib.

**Pull request merged**